### PR TITLE
WIP: Save new export state

### PR DESCRIPTION
**Pull request by benrudolph**
This is half done, but @snopoke / @czue wanted to get your opinion before i left. I'm working on saving the state of the Schema. My biggest concern so far with this approach is that if there's a bug in the way the export schema was generated initially, then we're kind of screwed unless we have a force rebuild function. this was the cleanest way i could think of doing it, but open to suggestions that would allow for more flexibility. it may be best to make an ExportDataSchema for each new export even though it's technically not necessary since each schema should be same based on the xmlns/case_type

### CODE REVIEW

File: corehq/couchapps/schemas_by_xmlns_or_case_type/views/view/map.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+function(doc) {
```
Comment (dimagimon) [inline corehq/couchapps/schemas_by_xmlns_or_case_type/views/view/map.js]: Missing name in function declaration.

File: corehq/apps/export/models/new.py [hunk 2]
```diff
@@ -254,6 +255,9 @@ class ExportDataSchema(Document):
     form xmlns or case type. It contains a list of ExportGroupSchema.
     """
     group_schemas = SchemaListProperty(ExportGroupSchema)
+
+    # A map of app_id to app_version. Represents the last time it saw an app and at what version
+    last_updated = DictProperty()
```
Comment (snopoke) [inline corehq/apps/export/models/new.py]: `last_updated` makes me think its going to be a date. Maybe `last_app_versions` or something?

File: corehq/apps/export/models/new.py [hunk 3]
```diff
@@ -318,8 +328,17 @@ def generate_schema_from_builds(domain, app_id, form_xmlns):
         :param unique_form_id: The unique identifier of the item being exported
         :returns: Returns a ExportDataSchema instance
         """
-        app_build_ids = get_built_app_ids_for_app_id(domain, app_id)
-        all_xform_conf = ExportDataSchema()
+        xform_schema_id = get_latest_export_schema_id(domain, 'FormExportDataSchema', form_xmlns)
+        if xform_schema_id:
+            all_xform_schema = FormExportDataSchema.get(xform_schema_id)
+        else:
+            all_xform_schema = FormExportDataSchema()
```
Comment (snopoke) [inline corehq/apps/export/models/new.py]: `all_xform_schema` -> `current_xform_schema`?

Comment (snopoke) [inline corehq/apps/export/models/new.py]: nit

**Comment by snopoke** [comment]
looks good

File: corehq/apps/export/models/new.py (hunk 2 above)
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: :bulb:

**Comment by benrudolph** [comment]
makes sense. any thoughts on whether it should be one doc per `xmlns`/`case_type` or per export instance?

File: corehq/apps/export/tests/test_export_data_schema.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import os
+from mock import Mock, patch
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_data_schema.py]: F401 'Mock' imported but unused

**Comment by benrudolph** [comment]
lots of goodies in this. probably easiest :fish:, but realize it's kind of long. amazing how a little bit of state can add so much complexity

**Comment by benrudolph** [comment]
opening against master

**Pull request closed without merging**